bob-apps/bin/bobp.rs

```diff
@@ -579,9 +651,69 @@ async fn put_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statis
     }
 }
 
+fn prepare_exist_keys(from: u64, count: u64, packet_size: u64, shuffle: bool) -> Vec<Vec<u64>> {
+    let mut keys = Vec::new();
+    let mut current_low = from;
+    for _ in 0..count {
+        let mut keys_portion: Vec<_> = 
+            (current_low..current_low + packet_size).collect();
+        if shuffle {
+            keys_portion.shuffle(&mut thread_rng());
+        }
+
+        keys.push(keys_portion);
+
+        current_low += packet_size;
+    }
+    keys
+}
+
+async fn exist_worker(net_conf: NetConfig, task_conf: TaskConfig, stat: Arc<Statistics>) {
+    let mut client = net_conf.build_client().await;
+
+    let options = task_conf.find_get_options();
+    let send_size_bytes = task_conf.packet_size * (task_conf.key_size as u64);
+    let request_count = task_conf.count / task_conf.packet_size;
+    let measure_time = task_conf.is_time_measurement_thread();
+
+    let keys = prepare_exist_keys(task_conf.low_idx, request_count, task_conf.packet_size, task_conf.is_random());
+    let iterator = Box::new(keys.into_iter());
+
+    for portion in iterator {
+        let keys = portion.iter().map(|key| 
+            BlobKey {
+                key: task_conf.get_proper_key(*key),
+            }).collect();
+        let request = Request::new(ExistRequest {
+            keys,
+            options: options.clone(),
+        });
+        let res = if measure_time {
+            let start = Instant::now();
+            let res = client.exist(request).await;
+            stat.save_single_thread_exist_time(&start.elapsed());
+            res
+        } else {
+            client.exist(request).await
+        };
+        stat.exist_size_bytes
+                    .fetch_add(send_size_bytes, Ordering::SeqCst);
+        match res {
+            Err(status) => stat.save_exist_error(status).await,
+            Ok(payload) => {
```

Comment:
There should be an additional counter, that counts the number of presented keys out of the total requested keys (it should also be printed to console)